packages/google-pubsub-microservice/src/GooglePubSubServer.ts

```diff
@@ -0,0 +1,85 @@
+import { EmittedMessage, GCPubSub, GooglePubSubOptions, PubSubFactory, Transport } from '@algoan/pubsub';
+import { Logger } from '@nestjs/common';
+import { CustomTransportStrategy, MessageHandler, Server } from '@nestjs/microservices';
+
+/**
+ * Google Pub Sub Server class extending NestJS Microservice strategy
+ */
+export class GCPubSubServer extends Server implements CustomTransportStrategy {
+  public gcClient!: GCPubSub;
+
+  protected readonly logger: Logger = new Logger(GCPubSubServer.name);
+
+  constructor(private readonly options?: GooglePubSubOptions) {
+    super();
+  }
+
+  /**
+   * Server listening method
+   */
+  public listen(callback: () => void): void {
+    const gcPubSub: GCPubSub = PubSubFactory.create({
+      transport: Transport.GOOGLE_PUBSUB,
+      options: this.options,
+    });
+
+    this.gcClient = gcPubSub;
+    const handlers: Promise<void>[] = [];
+
+    for (const subscriptionName of this.messageHandlers.keys()) {
+      this.logger.debug(`Registered new subscription "${subscriptionName}"`);
+
+      handlers.push(
+        gcPubSub.listen(subscriptionName, {
+          onMessage: this.handleMessage(subscriptionName),
+          onError: this.handleError.bind(this),
```

Comment:
this.handleError.bind(this) will return a new handleError each time.
define it outside of the for of code block and pass its reference to the onError and catch handler

Reply:
Found another way